File: terminal_colors.c
// compile with gcc -o tc terminal_colors.c
#include <stdio.h>
#include <stdlib.h>

int main () {
  printf("8 colors\n");
  for (int i=0;i<8;++i) {
    printf("\033[%dm\\033[%dm ",30+i,30+i);
  }
  printf("\033[0m\n");
  for (int i=0;i<8;++i) {
    printf("\033[%dm\\033[%dm ",40+i,40+i);
  }
  printf("\033[0m\n");
  
  printf("256 colors \"\\033[38;5;\033[31mcode\033[0mm\"\n");
  for (int i=0;i<16;++i) {
    
    for (int j=0;j<16;++j) {
      printf("\033[38;5;%dm%4d ",(i<<4)+j,(i<<4)+j);
    }
    printf("\033[0m\n");
  }
  printf("\033[0m\n");
  
  printf("256 colors \"\\033[48;5;\033[31mcode\033[0mm\"\n");
  for (int i=0;i<16;++i) {
    
    for (int j=0;j<16;++j) {
      printf("\033[48;5;%dm%4d ",(i<<4)+j,(i<<4)+j);
    }
    printf("\033[0m\n");
  }
  printf("\033[0m\n");
  return 0;
}

File: terminal_colors.rs
// compile with: "rustc terminal_colors.rs"
fn main() {
    println!("8 colors"); 

    for i in 0..8 {
        print!("\x1b[{}m\\x1b[{}m ",30+i,30+i);
    }
    println!("\x1b[0m");
    
    for i in 0..8 {
        print!("\x1b[{}m\\x1b[{}m ",40+i,40+i);
    }
    println!("\x1b[0m");

    println!("256 colors \"\\x1b[38;5;\x1b[31mcode\x1b[0mm\""); 
    for i in 0..16 {
        for j in 0..16 {
            print!("\x1b[38;5;{}m{:3} ",(i<<4)+j,(i<<4)+j);
        }
        println!("\x1b[0m");
    }
    println!("\x1b[0m");
    
    println!("256 colors \"\\x1b[48;5;\x1b[31mcode\x1b[0mm\""); 
    for i in 0..16 {
        for j in 0..16 {
            print!("\x1b[48;5;{}m{:3} ",(i<<4)+j,(i<<4)+j);
        }
        println!("\x1b[0m");
    }
    println!("\x1b[0m");
    

}